test/app-luatest/tap_test.lua

```diff
@@ -0,0 +1,633 @@
+      ---
+      path: //a
+      strict: true
+      expected: nil
+      got: cdata
+      ...
+    not ok - nil and box.NULL strict = true
+      ---
+      got: nil
+      expected: cdata
+      strict: true
+      ...
+    not ok - box.NULL and nil strict = true
+      ---
+      got: cdata
+      expected: nil
+      strict: true
+      ...
+    ok - {a = box.NULL} and {a = box.NULL} strict true
+        # check strict flag inheritance
+        1..2
+        not ok - {} and {a = box.NULL} strict = true
+          ---
+          strict: true
+          expected: key a
+          got: nil
+          ...
+        not ok - nil and box.NULL strict = true
+          ---
+          got: nil
+          expected: cdata
+          strict: true
+          ...
+        # check strict flag inheritance: end
+    not ok - failed subtests
+      ---
+      planned: 2
+      failed: 2
+      ...
+    # is_deeply: end
+not ok - failed subtests
+  ---
+  planned: 21
+  failed: 9
+  ...]])
+end
+
+g.test_tap_module_like_keyword = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'tap_test_tap_module_like_keyword.lua'
+    treegen.write_file(dir, script_name, [[
+local tap = require('tap')
+local test = tap.test('root test')
+test.trace = false
+test:plan(8)
+test:test('like', function(t)
+    t:plan(2)
+    t:like('abcde', 'cd', 'like(abcde, cd)')
+    t:unlike('abcde', 'acd', 'unlike(abcde, acd)')
+end)
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, [[
+TAP version 13
+1..8
+    # like
+    1..2
+    ok - like(abcde, cd)
+    ok - unlike(abcde, acd)
+    # like: end
+ok - like]])
+end
+
+-- Test, that in case of not strict comparison the order of
+-- arguments does not matter.
+g.test_tap_module_comparison_order_does_not_matter = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name =
+        'tap_test_tap_module_comparison_order_does_not_matter.lua'
+    treegen.write_file(dir, script_name, [[
+local tap = require('tap')
+local test = tap.test('root test')
+test.trace = false
+test:plan(2)
+test:is_deeply({1, 2, 3}, '200', "compare {1, 2, 3} and '200'")
+test:is_deeply('200', {1, 2, 3}, "compare '200' and {1, 2, 3}")
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, [[
+TAP version 13
+1..2
+not ok - compare {1, 2, 3} and '200'
+  ---
+  strict: false
+  expected: '200'
+  got:
+  - 1
+  - 2
+  - 3
+  ...
+not ok - compare '200' and {1, 2, 3}
+  ---
+  strict: false
+  expected:
+  - 1
+  - 2
+  - 3
+  got: '200'
+  ...]])
+end
+
+-- Finish root test. Since we used non-callback variant, we have
+-- to call check explicitly.
+g.test_tap_module_check = function()
```

Comment:
Nit: It seems, it is covered by `test_tap_module_subtests_without_callbacks`.

Reply:
So should I remove `test_tap_module_check` or `test_tap_module_subtests_without_callbacks`?

Reply:
The original test performs some post-actions for the sake of completeness, because there is a one main `test` object. Now we have separate test cases and this is irrelevant. So `test_tap_module_check` seems redundant, because there are the same checks are in `test_tap_module_subtests_without_callbacks`.

OTOH, I just observed that we don't call `:check()` for the top-level `test` object, only for nested ones. So, maybe, my comment is not entirely correct.

The proposal was to remove `test_tap_module_check`, but I've found that it has something that is missed in other cases.

I'll resolve this comment so.